pkg/deploy/generator/resources_dev.go

```diff
@@ -445,3 +445,25 @@ func (g *generator) devDiskEncryptionKeyVaultAccessPolicy() *arm.Resource {
 		DependsOn:  []string{fmt.Sprintf("[resourceId('Microsoft.Compute/diskEncryptionSets', %s)]", sharedDiskEncryptionSetName)},
 	}
 }
+
+const (
+	SharedMSIKeyVaultName       = "concat(take(resourceGroup().name,10), '" + SharedMSIKeyVaultNameSuffix + "')"
```

Comment:
Can we use all the available length fot the keyvault name? I believe 10 was used in case of shared kv just to keep it to 23.

Reply:
Taking this comment back, it should follow the same assumption as the other KV otherwise in the RP backend it may need an additional handling:- https://github.com/Azure/ARO-RP/blob/master/pkg/util/cluster/cluster.go#L218C2-L224C3